wordpress-mobile/WordPress-Android
# Onboarding Improvements: Epilogue Screen Select Site

Parent #14990

This PR enables selecting a site from within the newly designed `Login Epilogue` screen.

NOTE: Selecting a site from the newly design `Login Epilogue` screen will now always show the `Quick Start Prompt`, no matter whether the user selected `Show me around` or `No thanks`. So, if the user logs-out and logs-back-in afterwards, then selects the very same site, then the `Quick Start Prompt` will be shown again.

-----

To test:

Scenario: `Dismiss`
1. Go to `My Site` -> `App Settings` -> `Test feature configuration`.
2. Enable the `OnboardingImprovementsFeatureConfig` feature flag.
3. Log-out and log-in to the app again.
4. When on the `Login Epilogue` screen, select a site from the list of sites.
5. Make sure the new `Quick Start Prompt` is shown.
6. Dismiss the fullscreen dialog by clicking or swiping back.
7. Make sure the `My Site` screen is show and that the `Quick Start` section, with `Next Steps` and all is not shown. 

Scenario: `No thanks`
1. Go to `My Site` -> `App Settings` -> `Test feature configuration`.
2. Enable the `OnboardingImprovementsFeatureConfig` feature flag.
3. Log-out and log-in to the app again.
4. When on the `Login Epilogue` screen, select a site from the list of sites.
5. Make sure the new `Quick Start Prompt` is shown.
6. Click the `No thanks` button.
7. Make sure the `My Site` screen is shown and that the `Quick Start` section, with its `Next Steps` and all, is not shown.

Scenario: `Show me around`
1. Go to `My Site` -> `App Settings` -> `Test feature configuration`.
2. Enable the `OnboardingImprovementsFeatureConfig` feature flag.
3. Log-out and log-in to the app again.
4. When on the `Login Epilogue` screen, select a site from the list of sites.
5. Make sure the new `Quick Start Prompt` is shown.
6. Click the `Show me around` button.
7. Make sure the `My Site` screen is shown and that the `Quick Start` section, with its `Next Steps` and all, is shown.

-----

## Regression Notes
1. Potential unintended areas of impact

The existing `Quick Start` logic might be somehow affected by all the changes in this PR since there has been some updates to all that.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
